include/nuttx/wdt_notifier.h

```diff
@@ -0,0 +1,97 @@
+/****************************************************************************
+ * include/nuttx/wdt_notifier.h
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __INCLUDE_NUTTX_WDT_NOTIFIER_H
+#define __INCLUDE_NUTTX_WDT_NOTIFIER_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <nuttx/notifier.h>
+#include <nuttx/sched.h>
+
+#include <sys/types.h>
+
+#ifdef CONFIG_WATCHDOG_TIMEOUT_NOTIFIER
+
+/****************************************************************************
+ * Public Types
+ ****************************************************************************/
+
+enum wdt_type_e
+{
+  WDT_KEEPALIVE_BY_ONESHOT         =  0,
+  WDT_KEEPALIVE_BY_TIMER           =  1,
```

Comment:
where do you use this macro/enum

Reply:
First of all, there was an original mistake here, `WATCHDOG_NOTIFIER_ACTION` should have been macro defined as different keepalive actions according to `CONFIG_WATCHDOG_AUTOMONITOR_BY_`, presented in watchdowatch.h.I have fixed this mistake.

Then, based on these actions, the user can determine the mechanism by which the current watchdog keeps alive and take different actions accordingly.

for example:
```C
int user_wdt_timeout_callback(FAR struct notifier_block *nb, unsigned long action,
                         FAR void *data)
{
  UNUSED(data);
  UNUSED(nb);
  switch(action):
  {
   case WATCHDOG_KEEPALIVE_BY_ONESHOT:
           //do something
           break;
   case WATCHDOG_KEEPALIVE_BY_TIMER:
           //do something
           break;
   ...
  }
  printf("The callback function registered by the user is triggered!\n");
  return 0;
}
```